src/javascript/events/component-view.js

```diff
@@ -0,0 +1,67 @@
+/*global module, require */
+'use strict'; // eslint-disable-line strict
+
+const Core = require('../core');
+const getTrace = require('../../libs/get-trace');
+const utils = require('../utils');
+
+const decorateEventData = (eventData, viewedEl, opts) => {
+
+	const { trace, customContext } = getTrace(viewedEl);
+	let context = typeof opts.getContextData === 'function' ? opts.getContextData(viewedEl) : {};
+
+	if (typeof context !== 'object') {
+		context = {};
+	}
+
+	context.domPathTokens = trace;
+	context.url = window.document.location.href || null;
+
+	utils.assignIfUndefined(customContext, context);
+
+	eventData.context = context;
+};
+
+/**
+ * Listen for view events.
+ *
+ * @alias view#init
+ * @param {Object} opts - To set custom category[String], selector[String], getContextData[Function]
+ * @return {undefined}
+ */
+const init = (opts = {}) => {
+	if(!window.IntersectionObserver) {
```

Comment:
Is it worth adding somewhere to the readme that this feature requires `window.IntersectionObserver` in order to track the events? https://caniuse.com/#feat=intersectionobserver

In FT.com we are loading the polyfill in n-ui, so not a problem 😀

Reply:
Oh,That is a good point. I'm going to add the explanation.

Reply:
It should be added to the `origami.json.browserFeatures.required` array. --> https://origami.ft.com/docs/syntax/origamijson/

Reply:
@JakeChampion Thank you for telling me!